viewing_party/party.py

```diff
@@ -1,23 +1,131 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {
+        "title": title, 
+        "genre": genre,
+        "rating": rating
+    }
+
+    if None in movie.values():
+        return None
+
+    else:
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in list(user_data["watchlist"]):
+        if movie["title"] == movie_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    if not user_data["watched"]:
+        return 0
 
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    genre_list = []
+# I guess this works.. has functionality to deal with
+# any list of genres instead of hardcoding counters for each.
+# Maybe better with a helper function called get_genre_list()?

```

Comment:
Keep comments indented with the level of code they're at. It's difficult to read them when they're mixed into the code in a jaggy fashion.

Reply:
Also, cheer up! 😁 The dictionary method of tallying genres here is spot on! If you're concerned about it being a little noisy in the code here, you could write a helper to build this data structure, or look into other python standard modules the next time you need to do this.